### PR TITLE
Improve container widget

### PR DESCRIPTION
**Pull request by r-mzy47**
in Container Widget when the `child` parameter is null and `constraints` is not Tight, there is no need to add the [Align] widget when the `alignment` parameter is provided.

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

### CODE REVIEW

File: packages/flutter/lib/src/widgets/container.dart [hunk 1]
```diff
@@ -391,7 +391,7 @@ class Container extends StatelessWidget {
       );
     }
 
-    if (alignment != null)
+    else if (alignment != null)
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/container.dart:394]: can you please add `{ }` around the body of the `if` and `else if` to make this a little more readable?

Comment (r-mzy47) [reply packages/flutter/lib/src/widgets/container.dart:394]: Thanks!

**Review by goderbauer — APPROVED** [review]
LGTM

**Review by Piinks — APPROVED** [review]
Hey @r-mzy47 welcome! Thanks for the contribution!
LGTM

**Comment by goderbauer** [comment]
@r-mzy47 Looks like the tests got stuck on this PR and for some reason I cannot restart them. Could you push and empty commit to this PR or rebase it to the latest master to re-trigger the tests? Thanks and sorry for the inconvenience.

**Comment by r-mzy47** [comment]
@goderbauer I have pushed an empty commit. But still, there are some pending tests. In addition, there are some failures added. have I done something wrong?

**Comment by goderbauer** [comment]
Looks like its an unrelated failure. Should be fixed if you rebase this branch to the latest master.

**Review by goderbauer — APPROVED** [review]
LGTM

**Pull request merged**